plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererService.kt

```diff
@@ -894,3 +902,7 @@ data class ResponseContext(
 interface CodeWhispererCodeCompletionServiceListener {
     fun onSuccess(fileContextInfo: FileContextInfo) {}
 }
+
+interface CodeWhispererIntelliSenseOnHoverListener {
+    fun onEnter() {}
```

Comment:
do we need this default implementation or simply make it abstract?

Reply:
I belive in kotlin something like this means it's abstract? Not totally sure, but all the existing subscribers work like this.

Reply:
nah i think to make it abstract `fun onEnter()` is sufficient and we don't need `{}`

Reply:
unless we need an implementation of `CodeWhispererIntelliSenseOnHoverListener ` which does nothing